kernel/Kconfig

```diff
@@ -521,6 +521,18 @@ config ARCH_HAS_CUSTOM_BUSY_WAIT
 	  the provided k_busy_wait(), but instead must do something custom. It must
 	  enable this option in that case.
 
+if ARCH_HAS_CUSTOM_BUSY_WAIT
+config ARCH_HAS_CUSTOM_MIXED_BUSY_WAIT
```

Comment:
This isn't an ARCH feature, it's not even board or soc-specific.  It has to do with whether or not the clock used to drive a custom busy wait is stable and calibrated or not, which is an application-level configuration choice even on nRF5.

I think I'd prefer this be handled at the platform layer, which is the point of "custom busy wait" to begin with.  It's not clear to me that there will be much benefit to having complicated generic code for this in the (currently very simple) k_busy_wait() layer.